starknet_py/contract.py

```diff
@@ -590,51 +595,89 @@ def _make_functions(
 
         return repository
 
+    @staticmethod
+    async def _from_address_direct(
+        address: Address,
+        client: Client,
+    ) -> "Contract":
+        """
+        Creates a Contract instance directly from given address.
+        """
+        contract_class = await get_class_by_address(address=address, client=client)
+        return Contract(address=address, abi=contract_class.abi or [], client=client)
 
-class ContractFromAddressFactory:
-    def __init__(
-        self,
-        address: AddressRepresentation,
-        client: GatewayClient,
-        max_steps: int,
-        proxy_checks: List[ProxyCheck],
-    ):
-        self._address = address
-        self._client = client
-        self._max_steps = max_steps
-        self._proxy_checks = proxy_checks
-        self._processed_addresses = set()
+    @staticmethod
+    async def _from_address_proxy_step(
```

Comment:
I don't like that so much logic related to resolving proxy is leaking here. I think I would rather leave `ContractFromAddressFactory` (although name could be improved, like ContractAbiResolver) or create a separate module for that.